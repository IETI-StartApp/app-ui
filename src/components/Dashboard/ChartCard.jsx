import React from 'react';
import { useStyles } from './styles';
import { Card, CardContent, Typography, Box } from '@material-ui/core/';

export const ChartCard = ({ title, chart}) => {
    const classes = useStyles();
    
    return (
        <Card variant="outlined" className={classes.cardChart}>
            <CardContent>
                <Typography className={classes.mainColor} variant="button" display="block">{ title }</Typography>
                <Box p={3}>
                    {
                        chart
                    }
                </Box>
            </CardContent>
        </Card>
    );
}