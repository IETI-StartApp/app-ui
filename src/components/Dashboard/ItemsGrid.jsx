import React from 'react';
import { Card, CardContent, Typography, Avatar, Grid, CardActionArea } from '@material-ui/core/';
import { useStyles } from './styles';
import { useHistory } from "react-router";

export const ItemsGrid = ({ items }) => {
    const classes = useStyles();
    const history = useHistory() || [];

    return (
        <Grid container spacing={2} direction="row" justify="center" className={classes.items}>
            {
                items.map(item => (
                    <Grid item xs={12} sm={5} lg={3}>
                        <div style={{ display: "flex", alignItems: "center", justifyContent: "center" }}>
                            <Card variant="outlined" className={classes.dashItem}>
                                <Avatar className={classes.cardAvatar} variant="rounded">
                                    {item.icon}
                                </Avatar>
                                <CardActionArea onClick={() => { history.push(item.path) }}>
                                    <CardContent className={classes.itemContent}>
                                        <Typography variant="h4" align="center" className={classes.mainColor} >{item.number}</Typography>
                                        <Typography align="center">{item.description}</Typography>
                                    </CardContent>
                                </CardActionArea>
                            </Card>
                        </div>
                    </Grid>
                ))
            }
        </Grid>
    );
}