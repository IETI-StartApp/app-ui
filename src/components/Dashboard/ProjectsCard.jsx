import React from 'react';
import { Card, CardContent, Typography, GridList, GridListTile } from '@material-ui/core/';
import CardProject from '../ProjectItems/CardProject';
import { useStyles } from './styles';

export const ProjectsCard = ({ title, projects }) => {
    const classes = useStyles();

    return (
        <Card variant="outlined" className={classes.projects}>
            <CardContent>
                <Typography className={classes.mainColor} variant="button" display="block">{ title }</Typography>
                <div className={classes.rootProjects}>
                    <GridList className={classes.gridList}>
                        {projects.map((project) => (
                            <GridListTile className={classes.gridListTile}>
                                <CardProject project={project} />
                            </GridListTile>
                        ))}
                    </GridList>
                </div>
            </CardContent>
        </Card>
    );
}