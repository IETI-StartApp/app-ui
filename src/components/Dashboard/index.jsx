import React from 'react';
import Navigation from '../Navbar/Navigation';
import ChatBubbleOutlineIcon from '@material-ui/icons/ChatBubbleOutline';
import { Grid, Fab } from '@material-ui/core/';
import { useStyles } from './styles';
import { ItemsGrid } from './ItemsGrid';
import { ProjectsCard } from './ProjectsCard';
import { ChartCard } from './ChartCard';

export const Dashboard = ( { items, projects, chart, projectsTitle, chartTitle } ) => {
    const classes = useStyles();

    return (
        <>
            <Navigation />
            <div className={classes.offset}></div>
            <div className={classes.root}>
                <Grid container spacing={2} direction="col" justify="center" alignItems="center" className={classes.allItems}>
                    <Grid item xs={12}>
                        <ItemsGrid items={items} />
                    </Grid>
                    <Grid item xs={12}>
                        <Grid container spacing={2} direction="row" justify="center">
                            <Grid item xs={12} sm={10} lg={6}>
                                <ChartCard title= { chartTitle } chart = { chart } />
                            </Grid>
                            <Grid item xs={12} sm={10} lg={6}>
                                <ProjectsCard title={ projectsTitle } projects={ projects } />
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>
                <Fab aria-label="Chat" className={classes.fab}>
                    <ChatBubbleOutlineIcon />
                </Fab>
            </div>
        </>
    );
}