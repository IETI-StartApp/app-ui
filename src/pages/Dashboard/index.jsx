import React, {useState,useEffect} from 'react';
import { InvestorDashboard, EntrepreneurDashboard} from "../../components";
import { getUserByEmail } from '../../services/userServices';
import {Auth} from "../../authServices/firebase-config";
export const Dashboard = () => {
    
    const [role, setRole] = useState("");
    useEffect(async() => {
        const user = await getUserByEmail(Auth.currentUser.email)
        setRole(user.role);
    }, [role])
    return (
        <>
            {
                role === "INVESTOR" &&
                <InvestorDashboard />
            }
            {
                role === "ENTREPRENEUR" &&
                <EntrepreneurDashboard />
            }
        </>
    );
}
