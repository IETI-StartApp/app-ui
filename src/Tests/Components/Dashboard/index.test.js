import React from 'react';
import { Dashboard } from '../../../components/Dashboard/index';
import { shallow } from 'enzyme';

describe('Testing dashboard', () => {

    const wrapper = shallow(<Dashboard  items={[]}  projects={[]} projectsTitle="Projects Title" chartTitle="Chart Title" />)

    test('should load dashboard ', () => {
        expect(wrapper).toMatchSnapshot();
    });

})