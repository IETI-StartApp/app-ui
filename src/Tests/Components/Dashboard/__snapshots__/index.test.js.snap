// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing dashboard should load dashboard  1`] = `
<Fragment>
  <Navigation />
  <div
    className="makeStyles-offset-13"
  />
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="makeStyles-allItems-8"
      container={true}
      direction="col"
      justify="center"
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <ItemsGrid
          items={Array []}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          direction="row"
          justify="center"
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={6}
            sm={10}
            xs={12}
          >
            <ChartCard
              title="Chart Title"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={6}
            sm={10}
            xs={12}
          >
            <ProjectsCard
              projects={Array []}
              title="Projects Title"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Fab))
      aria-label="Chat"
      className="makeStyles-fab-12"
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(Fab))>
  </div>
</Fragment>
`;
